packages/server/controllers/epochs.ts

```diff
@@ -178,14 +192,25 @@ export const getSlotsByEpoch = async (req: Request, res: Response) => {
             clickhouseClient.query({
                 query: `
                         SELECT
-                            pd.f_val_idx,
-                            pd.f_proposer_slot,
-                            pd.f_proposed,
-                            pk.f_pool_name
+                            pd.f_val_idx AS f_val_idx,
+                            pd.f_proposer_slot AS f_proposer_slot,
+                            pd.f_proposed AS f_proposed,
+                            pk.f_pool_name AS f_pool_name,
+                            hd.f_block AS f_block,
+                            bm.f_attestations AS f_attestations,
+                            bm.f_sync_bits AS f_sync_bits,
+                            bm.f_deposits AS f_deposits,
+                            bm.f_attester_slashings AS f_attester_slashings,
+                            bm.f_proposer_slashings AS f_proposer_slashings,
+                            bm.f_voluntary_exits AS f_voluntary_exits
                         FROM
                             t_proposer_duties pd
                         LEFT OUTER JOIN
                             t_eth2_pubkeys pk ON pd.f_val_idx = pk.f_val_idx
+                        LEFT OUTER JOIN
+                            t_head_events hd ON CAST(pd.f_proposer_slot AS String) = CAST(hd.f_slot AS String)
+                        LEFT OUTER JOIN
+                            t_block_metrics bm ON CAST(pd.f_proposer_slot AS String) = CAST(bm.f_slot AS String)
```

Comment:
We don't need to cast to string there since both are uints already. This cast is unnecessary and is making the query much slower.

Reply:
cast has been removed